esmvaltool/diag_scripts/rainfarm/main.r

```diff
@@ -0,0 +1,128 @@
+# #############################################################################
+# rainfarm.r
+# Authors:       E. Arnone (ISAC-CNR, Italy)
+#	         J. von Hardenberg (ISAC-CNR, Italy) 
+# #############################################################################
+# Description
+# ESMValTool diagnostic calling the RainFARM library written in Julia (by von Hardenberg, ISAC-CNR, Italy).
+# RainFARM is a stochastic precipitation downscaling method, further adapted for climate downscaling.
+#  
+# Required
+# CDO
+# Julia language: https://julialang.org
+# RainFARM Julia library: https://github.com/jhardenberg/RainFARM.jl
+#
+# Optional 
+#
+# Caveats
+#
+# Modification history
+#    20180508-A_arnone_e: Conversion to v2.0  
+#    20170908-A_arnone_e: 1st github version  
+#
+# ############################################################################
+
+library(tools)
+library(yaml)
+
+# read parameters
+source('esmvaltool/diag_scripts/rainfarm/rainfarm_parameters.r')
```

Comment:
Indeed the path of main.r should be used to figure this out, I believe we already discussed at the workshop how to do this.

Reply:
Sorry guys, I had these open issues in my mind and should have written about those straight from the beginning. Answering @bjoernbroetz points:

1)&2). For us Julia is needed only for the rainfarm tool. @jhardenberg I think you would be very happy to have that supported directly in esmvaltool. Would it be a good idea to run it directly without the r wrapper?

3) relative paths: I now recall we found a solution, here it is again. These below are the arguments passed by esmvaltool to the diagnostic, number 6 will do. I will apply it to all our tools. 
[1] "/usr/lib/R/bin/exec/R"                                                                              
[2] "--slave"                                                                                            
[3] "--no-restore"                                                                                       
[4] "--slave"                                                                                            
[5] "--quiet"                                                                                            
[6] "--file=/work/users/arnone/ESMValTool2/ESMValTool/esmvaltool/diag_scripts/rainfarm/main.r"           
[7] "--args"                                                                                             
[8] "/work/users/arnone/esmvaltool_output/recipe_rainfarm_20181002_125522/run/rainfarm/main/settings.yml" 

4) we discussed that before: we have quite some parameters that should not be touched by the usual user but only by an advanced one. Including them in the recipe would make it quite unreadable so that's why we are using a parameter file (it's almost like having them hard coded but without the mess of being really hard coded). Here it is just a matter of a couple of further  "sensitive" parameters, in other tools it would be quite a few of them. Do you want them in the recipe in any case?